app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/addEditNotificationModal.jsx

```diff
@@ -241,32 +238,40 @@ const AddEditNotificationModal = ({
     },
   ];
 
-  const {
-    actionType,
-    notification: { sendCase, informOwner },
-  } = data;
+  const { actionType } = data;
 
-  const okButton = {
-    text: formatMessage(COMMON_LOCALE_KEYS.SAVE),
-    onClick: () => {
-      const modalName = `${messages.title.defaultMessage.replace(
-        /\{.*\}/i,
-        messages[actionType].defaultMessage,
-      )}`;
+  const submitActions = (formFieldValues) => {
+    const modalName = `${messages.title.defaultMessage.replace(
+      /\{.*\}/i,
+      messages[actionType].defaultMessage,
+    )}`;
 
-      handleSubmit(onSave(!isEditorShown))();
+    const {
+      informOwner,
+      sendCase,
+      enabled: switcher,
+      attributes: { length },
+    } = formFieldValues;
 
-      const eventParameters = {
-        modalName,
-        status: informOwner,
-        type: messages[sendCase].defaultMessage,
-        switcher,
-        number: isEditorShown ? attributesValue.length : undefined,
-      };
+    const eventParameters = {
+      modalName,
+      status: informOwner,
+      type: messages[sendCase].defaultMessage,
+      switcher,
+      number: isEditorShown ? length : undefined,
+    };
 
-      trackEvent(PROJECT_SETTINGS_NOTIFICATIONS_EVENTS.CLICK_SAVE_BUTTON_IN_MODAL(eventParameters));
+    trackEvent(PROJECT_SETTINGS_NOTIFICATIONS_EVENTS.CLICK_SAVE_BUTTON_IN_MODAL(eventParameters));
+    onSave(!isEditorShown)(formFieldValues);
```

Comment:
Seems like now there is no need to use high order function for `onSave` at `notifications.jsx`.
Please refactor it and simplify the call.